sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -274,27 +282,36 @@ export abstract class V4PositionManager {
       planner.addIncrease(options.tokenId, position.liquidity, amount0Max, amount1Max, options.hookData)
     }
 
+    let value: string = toHex(0)
+
     // If migrating, we need to settle and sweep both currencies individually
     if (isMint(options) && options.migrate) {
-      // payer is v4 positiion manager
-      planner.addSettle(position.pool.currency0, false)
-      planner.addSettle(position.pool.currency1, false)
-      planner.addSweep(position.pool.currency0, options.recipient)
-      planner.addSweep(position.pool.currency1, options.recipient)
+      if (options.useNative) {
+        // unwrap the exact amount needed to send to the pool manager
+        planner.addUnwrap(OPEN_DELTA)
+        // payer is v4 position manager
+        planner.addSettle(position.pool.currency0, false)
+        planner.addSettle(position.pool.currency1, false)
+        // sweep any leftover wrapped native that was not unwrapped
+        planner.addSweep(position.pool.currency0.wrapped, options.recipient)
+        planner.addSweep(position.pool.currency1, options.recipient)
+      } else {
+        // payer is v4 position manager
+        planner.addSettle(position.pool.currency0, false)
+        planner.addSettle(position.pool.currency1, false)
+        planner.addSweep(position.pool.currency0, options.recipient)
+        planner.addSweep(position.pool.currency1, options.recipient)
+      }
```

Comment:
these `if` and `else` blocks are so similar and most of the lines are the same, i'd rather it wasn't duplicating so much logic 

```suggestion
        // unwrap the exact amount needed to send to the pool manager
        if (options.useNative) planner.addUnwrap(OPEN_DELTA)

        // payer is v4 position manager
        planner.addSettle(position.pool.currency0, false)
        planner.addSettle(position.pool.currency1, false)
        // for native sweep leftover wrapped native that wasnt unwrapped
        planner.addSweep(options.useNative ? position.pool.currency0.wrapped : position.pool.currency0, options.recipient)
        planner.addSweep(position.pool.currency1, options.recipient)
```